app/src/main/java/com/duckduckgo/app/fire/AutomaticDataClearer.kt

```diff
@@ -171,13 +182,19 @@ class AutomaticDataClearer(
         }
     }
 
-    private fun shouldClearData(cleanWhenOption: ClearWhenOption, appUsedSinceLastClear: Boolean): Boolean {
+    private fun shouldClearData(cleanWhenOption: ClearWhenOption, appUsedSinceLastClear: Boolean, appIconChanged: Boolean): Boolean {
         Timber.d("Determining if data should be cleared for option $cleanWhenOption")
 
+        if (appIconChanged) {
```

Comment:
I like the way this is integrated into this class; feels like a natural place for it 👍